compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -1571,6 +1571,24 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                                 }) {
                                     continue;
                                 }
+
+                                // opaque types generated when desugaring an async function can have a single
+                                // use lifetime even if it is explicitly denied (Issue #77175)
+                                if let hir::Node::Item(item) = self.tcx.hir().get(parent_hir_id) {
+                                    if let hir::ItemKind::OpaqueTy(ref opaque) = item.kind {
+                                        if opaque.origin == hir::OpaqueTyOrigin::AsyncFn {
+                                            // We want to do this only if the liftime identifier is already defined
+                                            // in the async function that generated this. Otherwise it could be
+                                            // an opaque type defined by the developer and we still want this
+                                            // lint to fail compilation
+                                            for p in opaque.generics.params {
+                                                if defined_by.contains_key(&p.name) {
+                                                    continue 'lifetimes;
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
```

Comment:
```suggestion
                                // opaque types generated when desugaring an async function can have a single
                                // use lifetime even if it is explicitly denied (Issue #77175)
                                if let hir::Node::Item(hir::Item {
                                    kind: hir::ItemKind::OpaqueTy(hir::OpaqueTy {
                                        origin: hir::OpaqueTyOrigin::AsyncFn,
                                        ..
                                    }),
                                    ..
                                }) = self.tcx.hir().get(parent_hir_id) {
                                    // We want to do this only if the liftime identifier is already defined
                                    // in the async function that generated this. Otherwise it could be
                                    // an opaque type defined by the developer and we still want this
                                    // lint to fail compilation
                                    for p in opaque.generics.params {
                                        if defined_by.contains_key(&p.name) {
                                            continue 'lifetimes;
                                        }
                                    }
                                }
```

Reply:
Taken your suggestion with some minor changes because I need the opaque type while trying to prevent rightwards drift